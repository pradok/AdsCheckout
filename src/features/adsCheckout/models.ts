import {customerModels} from '../customer';

export type Product = {
  id?: number | string;
  price: number;
  quantity: number;
  priceReduced: number;
  title: string;
};

export type Discount = {
  premium?: customerModels.DiscountPremium;
  classic?: customerModels.DiscountClassic;
  standout?: customerModels.DiscountStandOut;
  rulesDescription: string[];
};
