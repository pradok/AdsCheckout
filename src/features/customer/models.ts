export type DiscountPremium = {
  perAd: {
    minQty: number;
    price: number;
  };
};

export type DiscountStandOut = {
  price: number;
};

export type DiscountClassic = {
  trigger: number;
};
