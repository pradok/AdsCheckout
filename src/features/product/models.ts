export type Product = {
  id: string;
  title: string;
  price: number;
};

export interface ProductCollectionState {
  collection: Product[];
  isLoading: boolean;
  isFetched: boolean;
}
export interface ProductAction {
  type: string;
  payload?: Product[];
}
