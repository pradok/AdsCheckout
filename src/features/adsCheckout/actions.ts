import {actionTypes as at} from './constants';

export const addToCart = (sku: string) => {
  return {
    type: at.CHECKOUT_ADD_TO_CART,
    payload: sku,
  };
};
