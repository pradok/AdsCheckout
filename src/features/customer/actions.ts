import { actionTypes as at } from './constants';
import { CustomerDiscounts } from './models';

export const fetchCustomerDiscounts = () => {
  return {
    type: at.CUSTOMER_DISCOUNTS_FETCH,
  };
};

export const fetchSuccess = (result: CustomerDiscounts) => {
  return {
    type: at.CUSTOMER_DISCOUNTS_FETCH_SUCCESS,
    payload: result,
  };
};

export const fetchError = (error: Error) => {
  return {
    type: at.CUSTOMER_DISCOUNTS_FETCH_ERROR,
    payload: error,
  };
};
