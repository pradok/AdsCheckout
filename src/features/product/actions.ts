import { actionTypes as at } from './constants';
import { Product } from './models';

export const fetchProducts = () => {
  return {
    type: at.PRODUCTS_COLLECTION_FETCH,
  };
};

export const fetchSuccess = (result: Product[]) => {
  return {
    type: at.PRODUCTS_COLLECTION_FETCH_SUCCESS,
    payload: result,
  };
};

export const fetchError = (error: Error) => {
  return {
    type: at.PRODUCTS_COLLECTION_FETCH_ERROR,
    payload: error,
  };
};
