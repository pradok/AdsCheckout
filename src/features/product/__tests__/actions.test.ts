import * as actions from '../actions';
import {Product} from '../models';

const products: Product[] = [
  {
    id: 'classic',
    title: 'Classic Ad',
    price: 269.99,
  },
  {
    id: 'standout',
    title: 'Standout Ad',
    price: 322.99,
  },
  {
    id: 'premium',
    title: 'Premium Ad',
    price: 394.99,
  },
];

const error: Error = {
  name: 'Error Name',
  message: 'Error fetching products',
};

describe('features -> product -> actions', () => {
  it('fetchProducts', () => {
    expect(actions.fetchProducts()).toMatchSnapshot();
  });
  it('fetchProducts', () => {
    expect(actions.fetchSuccess(products )).toMatchSnapshot();
  });
  it('fetchProducts', () => {
    expect(actions.fetchError(error)).toMatchSnapshot();
  });
});
