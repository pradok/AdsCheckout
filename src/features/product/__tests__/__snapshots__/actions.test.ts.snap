// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features -> product -> actions fetchProducts 1`] = `
Object {
  "type": "PRODUCTS/COLLECTION_FETCH",
}
`;

exports[`features -> product -> actions fetchProducts 2`] = `
Object {
  "payload": Array [
    Object {
      "id": "classic",
      "price": 269.99,
      "title": "Classic Ad",
    },
    Object {
      "id": "standout",
      "price": 322.99,
      "title": "Standout Ad",
    },
    Object {
      "id": "premium",
      "price": 394.99,
      "title": "Premium Ad",
    },
  ],
  "type": "PRODUCTS/COLLECTION_FETCH_SUCCESS",
}
`;

exports[`features -> product -> actions fetchProducts 3`] = `
Object {
  "payload": Object {
    "message": "Error fetching products",
    "name": "Error Name",
  },
  "type": "PRODUCTS/COLLECTION_FETCH_ERROR",
}
`;
